fmriprep/reports/core.py

```diff
@@ -22,90 +22,38 @@
+from nireports.assembler.report import Report
 
-# This patch is intended to permit fMRIPrep 20.2.0 LTS to use the YODA-style
-# derivatives directory. Ideally, we will remove this in 20.3.x and use an
-# updated niworkflows.
 
-
-class Report(_Report):
-    def _load_config(self, config):
-        from yaml import safe_load as load
-
-        settings = load(config.read_text())
-        self.packagename = self.packagename or settings.get("package", None)
-
-        # Removed from here: Appending self.packagename to self.root and self.out_dir
-        # In this version, pass reportlets_dir and out_dir with fmriprep in the path.
-
-        if self.subject_id is not None:
-            self.root = self.root / f"sub-{self.subject_id}"
-
-        if "template_path" in settings:
-            self.template_path = config.parent / settings["template_path"]
-
-        self.index(settings["sections"])
-
-
-#
-# The following are the interface used directly by fMRIPrep
-#
-
-
-def run_reports(
-    out_dir,
-    subject_label,
-    run_uuid,
-    config=None,
-    reportlets_dir=None,
-    packagename=None,
-):
-    """
-    Run the reports.
-
-    .. testsetup::
-
-    >>> copytree_or_skip("data/tests/work", testdir)
-    >>> (testdir / 'fmriprep').mkdir(parents=True, exist_ok=True)
-
-    .. doctest::
-
-    >>> run_reports(testdir / 'out', '01', 'madeoutuuid', packagename='fmriprep',
-    ...             reportlets_dir=testdir / 'work' / 'reportlets' / 'fmriprep')
-    0
-
-    """
-    return Report(
-        out_dir,
-        run_uuid,
-        config=config,
-        subject_id=subject_label,
-        packagename=packagename,
-        reportlets_dir=reportlets_dir,
-    ).generate_report()
-
-
-def generate_reports(
-    subject_list, output_dir, run_uuid, config=None, work_dir=None, packagename=None
-):
-    """Execute run_reports on a list of subjects."""
+def generate_reports(subject_list, output_dir, run_uuid, config=None, work_dir=None):
+    """Generate reports for a list of subjects."""
     reportlets_dir = None
     if work_dir is not None:
         reportlets_dir = Path(work_dir) / "reportlets"
-    report_errors = [
-        run_reports(
+
+    report_errors = []
+    for subject_label in subject_list:
+        entities = {}
+        entities["subject"] = subject_label
+
+        robj = Report(
             output_dir,
-            subject_label,
             run_uuid,
-            config=config,
-            packagename=packagename,
+            bootstrap_file=config,
             reportlets_dir=reportlets_dir,
+            plugins=None,
+            plugin_meta=None,
+            metadata=None,
+            **entities,
         )
-        for subject_label in subject_list
-    ]
 
-    errno = sum(report_errors)
+        # Count nbr of subject for which report generation failed
+        errno = 0
+        try:
+            robj.generate_report()
+        except:
+            errno += 1
```

Comment:
The point of this previously was to count errors that were added to the report, not count failures to generate reports. It seems that functionality was removed in the transition to nireports, or perhaps it didn't exist in MRIQC and that was used as the model for nireports.

@oesteban might be able to share the history here...

Reply:
Yes, apologies for the slow response.

Indeed, the new version of the report generation does not return the crashfiles count. It feels a bit clunky that pipeline errors (crashfiles) and report generation errors are mixed up together. For example, if tomorrow we figure (and now with fit+transform this scenario is more likely) a function mode where reports are not generated - we still want to know if crashfiles were generated.

So I would leave this as it stands now, but also add the counting of crashfiles after pipeline execution, so that we do not miss that error managing bit the report generation was doing for us.

Does that make sense?